dev/fattest.simplicity/src/componenttest/rules/repeater/JakartaEEAction.java

```diff
@@ -276,12 +276,10 @@ public static void transformApp(Path appPath, Path newAppPath, EEVersion eeVersi
      * @param transformationRulesAppend The map with the additional transformation rules to add
      */
     public static void transformApp(Path appPath, Path newAppPath, Map<String, String> transformationRulesAppend) {
-        if (RepeatTestFilter.isRepeatActionActive(EE9_ACTION_ID)) {
-            JakartaEE9Action.transformApplication(appPath, newAppPath, transformationRulesAppend);
-        } else if (RepeatTestFilter.isRepeatActionActive(EE10_ACTION_ID)) {
-            JakartaEE10Action.transformApplication(appPath, newAppPath, transformationRulesAppend);
-        } else if (RepeatTestFilter.isRepeatActionActive(EE11_ACTION_ID)) {
-            JakartaEE11Action.transformApplication(appPath, newAppPath, transformationRulesAppend);
+        RepeatTestAction action = RepeatTestFilter.getMostRecentRepeatAction();
+
+        if (action instanceof FeatureReplacementAction) {
```

Comment:
This should be an instanceof change of JakartaEEAction.  The point of having this common abstract class is not have have the logic in FeatureReplacementAction, but instead of have it in this class.